.gitattributes

```diff
@@ -0,0 +1,58 @@
+# Note that we have the line below just so that files whose extensions aren't
+# listed explicitly will be flagged by `check-gitattributes` as needing to be
+# added to this file
+* text=auto
```

Comment:
ok thinking on this one a bit more, I'm not sure we want this as the default, even if it is  primarily to ensure that we catch unlisted files in CI.  I believe this line will make it so that the first time someone checks in a file with an new file type, we will let git decide whether it is text, which is what we were trying to avoid by using the long list of extensions below.  We'll immediately complain in CI, but by this time git may have already mangled their binary file.  Probably safer to default to `binary` 🤔

Reply:
with `auto` I think you're at a bigger risk from text files seen as binary rather than the other way around. That said, I've not used this particular feature myself (we have .gitattributes to mark compiled JS and CSS as binary, that's it).

Reply:
No text files seen as binary is fine.  In that case, `git` won't touch the file when you commit it, and then we'll catch the lack of a pattern fro the new file in CI with the new `pre-commit` rule.  The other direction is the problem, because git could try to convert `CRLF` to `LF` in a binary file, which would be bad

Reply:
Fixed in latest commit